end_to_end_tests/functional_tests/generated_code_execution/test_docstrings.py

```diff
@@ -36,18 +38,57 @@ def get_section(self, header_line: str) -> List[str]:
       required: ["reqStr", "undescribedProp"]
 """)
 @with_generated_code_import(".models.MyModel")
-class TestSchemaDocstrings:
+class TestSchemaDocstringsDefaultBehavior:
     def test_model_description(self, MyModel):
         assert DocstringParser(MyModel).lines[0] == "I like this type."
 
-    def test_model_properties(self, MyModel):
+    def test_model_properties_in_model_description(self, MyModel):
         assert set(DocstringParser(MyModel).get_section("Attributes:")) == {
             "req_str (str): This is necessary.",
             "opt_str (Union[Unset, str]): This isn't necessary.",
             "undescribed_prop (str):",
         }
 
 
+@with_generated_client_fixture(
+"""
+components:
+  schemas:
+    MyModel:
+      description: I like this type.
+      type: object
+      properties:
+        prop1:
+          type: string
+          description: This attribute has a description
+        prop2:
+          type: string  # no description for this one
+      required: ["prop1", "prop2"]
+""",
+config="docstrings_on_attributes: true",
+)
+@with_generated_code_import(".models.MyModel")
+class TestSchemaWithDocstringsOnAttributesOption:
+    def test_model_description_is_entire_docstring(self, MyModel):
+        assert MyModel.__doc__.strip() == "I like this type."
+
+    def test_attrs_have_docstrings(self, generated_client: GeneratedClientContext):
+        # A docstring that appears after an attribute is *not* stored in __doc__ anywhere
+        # by the interpreter, so we can't inspect it that way-- but it's still valid for it
+        # to appear there, and it will be recognized by documentation tools. So we'll assert
+        # that these strings appear in the source code. The code should look like this:
+        #     class MyModel:
+        #         """I like this type."
+        #         prop1: str
+        #         """This attribute has a description"
+        #         prop2: str
```

Comment:
this sample doesn't work as is (was able to validate the regex assertions in the test after making the change with the missing quotations though)
```suggestion
        #    class MyModel:
        #        """I like this type."""
        #        prop1: str
        #        """This attribute has a description"""
        #        prop2: str
```

Reply:
@damola-benchling I agree that the commented code should be fixed, but I'm not sure what you mean by "was able to validate the regex assertions in the test after making the change". The test assertions are not being made against the example code in these comments, they're being made against the actual generated code.

Reply:
> The test assertions are not being made against the example code in these comments, they're being made against the actual generated code

True, but I assume (hope?) that the two should be equivalent (and pass the regex match) for maintainability’s sake.

Reply:
Yes, I understand that they should be consistent and I am going to fix the commented code. I just was (& am) unclear on the literal meaning of what you said about "validating the regex assertions in the test."

Reply:
Oh sure, I literally copied the sample reference (expected generated code) as a string and also copied the regexes and run it manually in a repl 
(I thought the regex pattern was interesting and just wanted to make sure I was reading it correctly)